src/libmongoc/tests/test-mongoc-loadbalanced.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2020-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mongoc/mongoc.h"
+#include "mongoc/mongoc-client-session-private.h"
+#include "test-conveniences.h"
+#include "test-libmongoc.h"
+#include "TestSuite.h"
```

Comment:
These tests are not run on Evergreen. I have run this locally to verify, but will add to Evergreen as part of CDRIVER-4062.

Reply:
What are these tests/where are they required in the spec?

Reply:
These are not required by the spec. I would like to include them as a simple validation of the implementation up to this point.

Load balancer support is still incomplete. My process has been run a narrowly scoped failing test, make the minimal behavior change, then refactor. ("red" => "green" => "refactor").

The server selection JSON spec tests verify server selection works.

The SDAM JSON tests verify that SDAM is disabled.

`/loadbalanced/connect/pooled` and `/loadbalanced/connect/single` verify that sending a command uses the server description tied to the connection. Without the recent fix of CDRIVER-3653, those tests fail.

`/loadbalanced/cooldown_is_bypassed/single` verifies behavior specific to the C implementation. Connection creation in single-threaded goes through the same machinery as SDAM. Without an additional change to bypass the 5 second cooldown, a network error could wait 5 seconds before successfully reconnecting.

`/loadbalanced/server_selection_establishes_connection/single` verifies a requirement from wrapping drivers: successful server selection implies a connection was established to a server and is ready for commands.

Reply:
Sounds good to me.